python/TestHarness/util.py

```diff
@@ -717,21 +717,24 @@ def getOutputFromFiles(tester, options):
     for file_path in output_files:
         with open(file_path, 'r') as f:
             file_output += "#"*80 + "\nOutput from " + file_path \
-                           + "\n" + "#"*80 + "\n" + readOutput(f, None, options)
+                           + "\n" + "#"*80 + "\n" + readOutput(f, None, options, max_size=tester.specs['max_buffer_size'])
     return file_output
 
 # This function reads output from the file (i.e. the test output)
 # but trims it down to the specified size.  It'll save the first two thirds
 # of the requested size and the last third trimming from the middle
-def readOutput(f, e, options, max_size=100000):
+def readOutput(f, e, options, max_size=None):
+    if max_size is None:
+        max_size = 100000
```

Comment:
Here it is... Instead of using None as the default, why not just put this value into the options directly?

Reply:
@milljm's explanation to me was that there are two other places in the TestHarness where `readOutput` is called and the `max_size` is not specified in those calls. So if `None` is not used, then we have to specify the default number in both the `params.addParam` default as well as in the function definition line. If we ever want to change the default `max_size`, we would then have to remember to change the number in two places. By using `None`, we can set it once here in the function body

Reply:
Sounds good. Now we just need to document what `None` does in the doc string. It's not clear what it does (it is after all the default).